### PR TITLE
Requesting a PR to master

### PR DESCRIPTION
**Pull request by mquadrant**
In this pull request: are methods written either on String or Number prototype to perform some manipulation on a string or number.

- hasVowels() returns true if the string contains at least a vowel letter. 
- toUpper()  returns the passed string in its upper case. 
- toLower() returns the passed string in its lower case.
- ucFirst() returns the passed string with only the first letter capitalize.
- isQuestion() returns true if the string passed is a question. 
- words() return an array of word broken from a string.
- wordCount() return the length of the words from a string. 
- toCurrency() return a currency equivalent of a given number. 
- fromCurrency() return a number equivalent from a given currency. 
- inverseCase() return a string with its case inverted.
- alternatingCase() return a string with the first letter a small case and the next capital case and so on.
- numberWords() return words equivalent of the number passed.
- isDigit() return a true if the passed number is a digit.

Screenshot 2019-05-15 at 17 15 11
Above is the test coverage

kindly review please.

### CODE REVIEW

File: method/methods.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//hasvowel return true if the string contain a vowel
+String.prototype.hasVowel = function () {
+    return /[aeiouAEIOU]/.test(this);
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain what the pattern is matching.

File: method/methods.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+//toUpper return the string in capital case
+String.prototype.toUpper = function () {
+    return this.replace(/[a-z]/g, function (str) {
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain what the pattern is matching.

Comment (mogbeyi-david) [reply method/methods.js]: Better to refer to the parameter passed to the function as **string** not str

File: method/methods.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        return String.fromCharCode(str.charCodeAt(0) - 32);
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain this line of code

File: method/methods.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    });
+}
+
+//toLower return the string in lower case
+String.prototype.toLower = function () {
+    return this.replace(/[A-Z]/g, function (str) {
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain what the pattern is matching.

File: method/methods.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        return String.fromCharCode(str.charCodeAt(0) + 32);
+    });
+}
+
+//ucFirst return the string with the first character in Upper case
+String.prototype.ucFirst = function () {
+    return this.replace(/[A-Za-z]/, function (str) {
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain what the pattern is matching.

File: method/methods.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+//ucFirst return the string with the first character in Upper case
+String.prototype.ucFirst = function () {
+    return this.replace(/[A-Za-z]/, function (str) { 
+        return str.toUpper()});
+}
+
+String.prototype.isQuestion = function () {
+    return /\?$/.test(this);
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain what the pattern is matching.

File: method/methods.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    return this.match(/\b[\w-]+\b/g);
+}
+
+String.prototype.wordCount = function () {
+    return this.words().length;
+}
+
+Number.prototype.toCurrency = function () {
+    var separate = this.toString().split('.');
+    separate[0] = separate[0].replace(/\B(?=(?:\d{3})+$)/g, ',');
```
Comment (mogbeyi-david) [inline method/methods.js]: Add a comment here to explain what the pattern is matching.

File: .gitignore [hunk 8]
```diff
@@ -1 +1,2 @@
-node_modules
+/node_modules
```
Comment (mogbeyi-david) [inline .gitignore:1]: Also, add a screenshot of your test coverage

File: test/methods.test.js [hunk 9]
```diff
@@ -1,6 +1,109 @@
...
+describe("Test for hasVowel() method in the String Prototype", () => {
+    it("should return 'true' if the string contains vowels", () => {
         var result = "Going home".hasVowel();
         expect(result).toBeTruthy();
     });
+});
+
+describe("Test for toUpper() method in String Prototype", () => {
+    it("should return the passed-in string with all characters in upper cases", () => {
+        var result = "Going 3467 home".toUpper();
```
Comment (mogbeyi-david) [inline test/methods.test.js]: Always use single quotes for Javascript strings

**Review by mogbeyi-david — CHANGES_REQUESTED** [review]

File: test/methods.test.js (hunk 9 above)
Comment (mquadrant) [reply test/methods.test.js]: okay. it has been effected.

File: .gitignore (hunk 8 above)
Comment (mquadrant) [reply .gitignore:1]: Screenshot 2019-05-15 at 16 30 38
Test coverage

File: method/methods.js (hunk 7 above)
Comment (mquadrant) [reply method/methods.js]: The pattern is matching set of three consecutive digit and same is bounded at the left by a word character, and also do not involve in the consumption of character.

File: method/methods.js (hunk 5 above)
Comment (mquadrant) [reply method/methods.js]: Replace method takes in a regex parameter that matches a single capital letter and small letter starting from the beginning of the string.

File: method/methods.js (hunk 6 above)
Comment (mquadrant) [reply method/methods.js]: The regular expression pattern is matching question mark that appear at the very end of the string.

File: method/methods.js (hunk 4 above)
Comment (mquadrant) [reply method/methods.js]: regex pattern matches all capital letter in the string

File: method/methods.js (hunk 3 above)
Comment (mquadrant) [reply method/methods.js]: Subtracting 32 from the base 10 equivalent of the passed character string would produce the lower case equivalent. And then convert resulting base 10 back to character.

File: method/methods.js (hunk 2 above)
Comment (mquadrant) [reply method/methods.js]: Regex pattern matches all small letters in the string.
Yes **string** not str, noted thanks.

File: method/methods.js (hunk 1 above)
Comment (mquadrant) [reply method/methods.js]: The regular expression pattern matches all vowel both capital and small letter.

**Pull request merged**